src/main/reseal.c

```diff
@@ -299,207 +295,130 @@ int main(int argc, char **argv)
-char *tempIn = outPath;
-char *tempOut = inPath;
-
-// allocates the updated outPath_size memory after the swap
-outPath_size = strlen(tempOut) + 1;
-if (outPath_size > 1)
-{
-  tempOut = malloc(outPath_size * sizeof(char));
-  memcpy(tempOut, inPath, outPath_size);
-}
-
-*/
+  
+  uint8_t *seal_output = NULL;
+  size_t seal_output_len = 0;
 
   // Call top-level "kmyth-seal" function
-  if (tpm2_kmyth_seal(input, input_length, &output, &output_length,
-                           (uint8_t *) authString, auth_string_len,
-                           (uint8_t *) ownerAuthPasswd, oa_passwd_len,
-                           pcrs, (size_t)pcrs_len, cipherString, expected_policy,
-                           bool_trial_only))
+if (tpm2_kmyth_seal(unseal_output, unseal_output_len, &seal_output, &seal_output_len,
+                      (uint8_t *) authString,
+                      auth_string_len,
+                      (uint8_t *) ownerAuthPasswd,
+                      oa_passwd_len,
+                      pcrs,
+                      (size_t) pcrs_len,
+                      cipherString,
+                      expected_policy,
+                      bool_trial_only))
   {
     kmyth_log(LOG_ERR, "kmyth-seal error ... exiting");
     kmyth_clear(authString, auth_string_len);
     kmyth_clear(ownerAuthPasswd, oa_passwd_len);
+    kmyth_clear_and_free(unseal_output, unseal_output_len);
     free(pcrs);
-    free(outPath);
-    free(output);
+    free(unseal_output);
     return 1;
   }
 
   kmyth_clear(authString, auth_string_len);
   kmyth_clear(ownerAuthPasswd, oa_passwd_len);
 
 
-  if (write_bytes_to_file(outPath, output, output_length))
+   // rename input file to <input filename>.orig to preserve it
+  char * renamePath = malloc(strlen(inPath) + strlen(".orig") + 1);
+  strncpy(renamePath, inPath, strlen(inPath));
+  strncat(renamePath, ".orig", 5);
+  if (!stat(renamePath, &st) && !forceOverwrite)
   {
-    kmyth_log(LOG_ERR, "error writing data to .ski file ... exiting");
-    free(outPath);
-    free(output);
-    free(pcrs);
+    kmyth_log(LOG_ERR,
+              "output filename (%s) already exists ... exiting",
+              renamePath);
     return 1;
   }
-
-/*
-  // removes the unsealed file
-  if(remove(tempIn) == 0){
-    kmyth_log(LOG_DEBUG, "%s successfully deleted", tempIn);
+  
+  if (rename((const char *) inPath, (const char *) renamePath) != 0)
+  {
+    kmyth_log(LOG_ERR, "renaming of input file failed ... exiting");
+    free(seal_output);
+    free(renamePath);
+    return 1;
   }
-  else{
-    kmyth_log(LOG_ERR, "unsealed file failed to be deleted..");
+  free(renamePath);
+  if (write_bytes_to_file(outPath, seal_output, seal_output_len))
+  {
+    kmyth_log(LOG_ERR, "error writing data to .ski file ... exiting");
+    free(seal_output);
   }
-*/
-
-
-
-  free(pcrs);
-  free(outPath);
-  free(output);
-  //free(tempOut);
 
+  kmyth_clear_and_free(unseal_output, unseal_output_len);
```

Comment:
This is very minor, but, since we no longer need `unseal_output` after the the tpm2_kmyth_seal() call, I think we could move this earlier.